HeterogeneousCore/AlpakaInterface/README.md

```diff
@@ -4,3 +4,310 @@ This package only depends on the `alpaka` header-only external library, and
+    allocates a one-dimensional buffer holding `size` objects of type `T` in the
+    global memory of device.
+
+These function allocate the buffers in device global memory, using potentially 
+blocking operations, without any caching.
+
+The memory is not initialised.
+
+
+### Allocating queue-ordered device buffers
+
+These helper functions allocate zero-dimensional (scalars) and one-dimensional
+(array) alpaka queue-ordered buffers in device global memory:
+
+  - `auto make_device_buffer<T>(queue)` (where `T` is not an array)
+    allocates a zero-dimensional buffer holding a single object of type `T` in
+    the global memory of `device`;
+
+  - `auto make_device_buffer<T[N]>(queue)`
+    allocates a one-dimensional buffer holding `N` objects of type `T` in the
+    global memory of `device`;
+
+  - `auto make_device_buffer<T[]>(queue, size)`
+    allocates a one-dimensional buffer holding `size` objects of type `T` in the
+    global memory of device.
+
+These functions use queue-ordered memory allocations.
+Queue-ordered device buffers are available once all the operations that have
+been enqueued in the given queue up to that point have completed.
+In a similar manner, when the buffers go out of scope or are destroyed, the
+buffer memory and the underlying objects are kept alive and remain valid until
+all the operations that have been enqueued in the given queue up to that point
+have completed.
+
+These function allocate the buffers in device global memory, potentially using
+caching to reduce the allocation overhead.
+
+The memory is not initialised.
+
+
+## Host memory views
+
+These type aliases simplify the use of zero-dimensional (scalars) and
+one-dimensional (array) alpaka host views:
+
+  - `host_view<T>` (where `T` is not an array)
+    represents a zero-dimensional view over a single object of type `T` in host
+    memory;
+
+  - `host_view<T[N]>`
+    represents a one-dimensional view over `N` objects of type `T` in host
+    memory;
+
+  - `host_view<T[]>`
+    represents a one-dimensional view over a number of objects of type `T` in
+    host memory determined at construction time.
+
+
+## Instantiatig host views
+
+These helper functions instantiate zero-dimensional (scalars) and one-dimensional
+(array) alpaka views over existing objects in host memory:
+
+  - `auto make_host_view<T>(T& data)` (where `T` is not an array)
+    instantiates a zero-dimensional view over the object `data` in host memory;
+
+  - `auto make_host_view<T[N]>(T[N]& data)`
+    instantiates a one-dimensional view over an array `data` in host memory;
+
+  - `auto make_host_view<T[]>(T* data, size)`
+    instantiates a one-dimensional view over an array starting at `data` with
+    `size` elements in host memory;
+
+
+## Device memory views
+
+These type aliases simplify the use of zero-dimensional (scalars) and
+one-dimensional (array) alpaka device views:
+
+  - `device_view<T>` (where `T` is not an array)
```

Comment:
Should these have the `Device` template argument? (like `device_buffer` has)

```suggestion
  - `device_view<Device, T>` (where `T` is not an array)
```

Reply:
I think so, yes.
Thanks, I will add them.